diff --git a/gin/spec/runners/integration.lua b/gin/spec/runners/integration.lua
index 5ef537a..21ec8f9 100644
--- a/gin/spec/runners/integration.lua
+++ b/gin/spec/runners/integration.lua
@@ -85,7 +85,7 @@ local function hit_server(request)
     local full_url = url.build({
-        scheme = 'http',
+        scheme = request.scheme or 'http',
         host = '127.0.0.1',
         port = Gin.settings.port,
         path = request.path,
